tests/cmake.py

```diff
@@ -136,19 +150,80 @@ def cmake(cwd, targets, options=None):
 
     print("\n{} > {}".format(cwd, " ".join(configcmd)), flush=True)
     try:
+        failed = False
         subprocess.run(configcmd, cwd=cwd, env=env, check=True)
     except subprocess.CalledProcessError:
+        failed = True
+    if failed:
         pytest.fail("cmake configure failed")
```

Comment:
I'm not sure what the reasoning behind this `failed = False` pattern is?  Am I missing a reason why you can't do
```
try:
    subprocess.run(...)
except subprocess.CalledProcessError:
    pytest.fail(...)
```
?  You're clearly just adding this, so there must be a reason.  To me it looks very weird though.